libsolidity/ast/Types.cpp

```
@@ -152,23 +154,31 @@ void StorageOffsets::computeOffsets(TypePointers const& _types)
 		{
 			// would overflow, go to next slot
 			++slotOffset;
+			++storageSize;
```

Wouldn't it be simpler to just subtract `_baseSlot` from `slotOffset` at the end?

---

I was kinda undecided about which way to do. The value is needed in two asserts and one assignment, so I just used the variable. I see no problem changing it to just subtracting though.

---

You don't actually need an assert against `storageSize` inside the loop. The one for `slotOffset` is there only because we're converting it to `u256` and that would silently truncate a value that's too big. And the opposite is true at the end of the function. You're only converting `storageSize` there.

And overall, the current approach duplicates the logic (however simple it may be) which is just gives you one more place for a mistake to creep in :)

---

Right. Fixed.